File: 980_div_1/a.cpp
#include <algorithm>
#include <stdio.h>
#include <vector>
using namespace std;

vector< pair<int, int> > a;
int t,n;

bool cmp(pair<int, int> l, pair<int, int> r) {
  if (min(l.first, l.second) == min(r.first, r.second)) {
    return max(l.first, l.second) < max(r.first, r.second);
  }

  return min(l.first, l.second) < min(r.first, r.second);
}

int main() {
  scanf("%d", &t);

  while (t--) {
    scanf("%d", &n);
    a.clear();

    for (int i=0; i<n; i++) {
      int x,y;
      scanf("%d %d", &x,&y);

      a.push_back(make_pair(x,y));
    }

    sort(a.begin(), a.end(), cmp);

    for(int i=0; i<n; i++)
      printf("%d %d ", a[i].first, a[i].second);
    printf("\n");
  }
  return 0;
}

File: 980_div_1/a.py
import functools

t = int(input())

def cmp(a, b):
    if min(a[0],a[1]) == min(b[0],b[1]):
        return max(a[0],a[1]) - max(b[0],b[1])
    return min(a[0],a[1]) - min(b[0],b[1])

for _ in range(t):
    n = int(input())
    a = []

    for i in range(n):
        x,y = input().split()
        a.append((int(x), int(y)))

    a.sort(key=functools.cmp_to_key(cmp))

    for i in range(n):
        print(a[i][0], " ", a[i][1], " ", end='', sep='')
    print();
